src/hooks/useSetupEngineManager.ts

```diff
@@ -43,7 +43,7 @@ export function useSetupEngineManager(
     engineCommandManager.pool = settings.pool
   }
 
-  useLayoutEffect(() => {
+  const startEngineInstance = () => {
```

Comment:
I'm not certain of the performance implications: should this use [`useCallback` to cache the callback between rerenders](https://react.dev/reference/react/useCallback)?

Reply:
This was part of a rebase; someone else added this :shrug:

Reply:
Yeah absolutely, no worries